brayns/common/PropertyMap.h

```diff
@@ -30,199 +30,227 @@
 namespace brayns
 {
 /**
- * Container class for holding properties that are mapped by name to a supported
- * C++ type and their respective value.
+ * Defines a single property which always has a name, a label for e.g. UIs
+ * and a typed value. Additionally, a user-defined minimum and maximum value
+ * range can be specified, otherwise the types' limits apply.
  */
-class PropertyMap
+struct Property
 {
-public:
-    PropertyMap() = default;
+    enum class Type
+    {
+        Int,
+        Double,
+        String,
+        Bool,
+        Vec2i,
+        Vec2d,
+        Vec3i,
+        Vec3d,
+        Vec4d
+    };
 
-    /**
-     * Defines a single property which always has a name, a label for e.g. UIs
-     * and a typed value. Additionally, a user-defined minimum and maximum value
-     * range can be specified, otherwise the types' limits apply.
-     */
-    struct Property
+    struct UserInfo
     {
-        enum class Type
-        {
-            Int,
-            Double,
-            String,
-            Bool,
-            Vec2i,
-            Vec2d,
-            Vec3i,
-            Vec3d,
-            Vec4d
-        };
-
-        template <typename T>
-        void assertValidType() const
+        UserInfo()
```

Comment:
Why is it named UserInfo? Do you mean UsageInfo?

Reply:
Called it MetaData now